chainer/utils/type_check.py

```diff
@@ -7,6 +7,9 @@
 
 from chainer import cuda
 
+if cuda.available:
+    import cupy
```

Comment:
use `cuda.cupy`. It is available regardless of availability of `cupy` (i.e. the value of `cuda.available`).